pandas/core/arrays/arrow/array.py

```diff
@@ -626,20 +626,22 @@ def __setstate__(self, state) -> None:
 
     def _cmp_method(self, other, op):
         pc_func = ARROW_CMP_FUNCS[op.__name__]
-        try:
+        if isinstance(other, (ArrowExtensionArray, np.ndarray, list, BaseMaskedArray)):
```

Comment:
Is it possible to split the cases when `ArrowNotImplementedError` vs `ArrowInvalid` is raised? I'm a little nervous of restricting the types of `other`?

Additionally maybe `self._box_pa` should be called first before going down the array like vs scalar path

Reply:
this basically restores the logic we had before the last refactor that introduced the regression. I think calling box_pa on everything raised an exception, but can retry

Reply:
Okay if this was the logic before I'm okay having this as a patch for 2.1.1, thought I think this should be more generic in the future.